modules/nextflow/src/main/groovy/nextflow/executor/SimpleFileCopyStrategy.groovy

```diff
@@ -159,23 +159,32 @@ class SimpleFileCopyStrategy implements ScriptFileCopyStrategy {
      */
     @Override
     String getUnstageOutputFilesScript(List<String> outputFiles, Path targetDir) {
-        final patterns = normalizeGlobStarPaths(outputFiles)
-        // create a bash script that will copy the out file to the working directory
-        log.trace "Unstaging file path: $patterns"
 
-        if( !patterns )
+        if( !outputFiles )
             return null
 
         final escape = new ArrayList(outputFiles.size())
-        for( String it : patterns )
-            escape.add( Escape.path(it) )
+        for( String it : outputFiles ){
+            String escaped = Escape.path( it, true )
+            //Bash glob behaves different than Java's Glob, if the path starts with **
+            //https://unix.stackexchange.com/questions/49913/recursive-glob
+            if( escaped.startsWith("**") && escaped.size() > 2 ) {
+                escaped = "**/*" + escaped.substring(2)
+            }
+            escape.add( escaped )
+        }
 
         final mode = stageoutMode ?: ( workDir==targetDir ? 'copy' : 'move' )
         return """\
             IFS=\$'\\n'
-            for name in \$(eval "ls -1d ${escape.join(' ')}" | sort | uniq); do
+            shopt -s globstar extglob || true
+            pathes=`ls -1d ${escape.join(' ')} | sort | uniq`
+            shopt -u globstar extglob || true
+            set -f
+            for name in \$pathes; do
```

Comment:
My understanding is this prevents the wildcards expansion. Why is needed? they should have already been resolved by the `ls -1d ` command

Reply:
You are right, `set -f` prevents it. This is needed, if file names contain a special character like a question mark, which would otherwise be resolved again. There is one test case which fails otherwise.

Reply:
Sorry, not fully understanding this. Can you provide a minimal example NF script showing how this works?

Reply:
Sure, the following process should only find one file - the one with a question mark in the file name.

```
process a {

    echo true

    output:
    file("ab\\?.txt") into found

    """
    touch abc.txt
    touch ab\\?.txt
    ls
    """

}
found.view()
```

Don't disable wildcard expansion would lead to a new expansion in the for loop and once again in the copy/move method. Accordingly, both files will be moved/copied.

Reply:
Hi @pditommaso, I’m not sure whether this helped you understand my contribution or you need an additional example. I look forward to your feedback.

Reply:
I see two problems: 1) handling correctly `glob:false` 2) handle `**` wildcard 

For 1, I think it could be addressed replacing `Escape.path` with `Escape.wildcard` when `glob==false` here 

https://github.com/nextflow-io/nextflow/blob/c405f9ebe778aef6113afdc0261faadf2e2f98cc/modules/nextflow/src/main/groovy/nextflow/executor/SimpleFileCopyStrategy.groovy#L171-L171

For 2, I think it would be preferable to keep eval subshell, I mean having 

```
for name in \$(eval "ls -1d ${escape.join(' ')}" | sort | uniq); do
... 
```